src/controller/handler/base.agent.response.handler.ts

```diff
@@ -0,0 +1,27 @@
+import { AxiosRequestConfig } from 'axios';
+import { ProtocolException } from 'protocol-common/protocol.exception';
+import { IAgentResponseHandler } from './agent.response.handler';
+
+/*
+    This class contains functions found in derived classes.  This class should never
+    be allocated directly, create a derived class.
+*/
+export abstract class BaseAgentResponseHandler implements IAgentResponseHandler {
+
+    public createHttpRequest(url: string, adminApiKey: string, token?: string): AxiosRequestConfig {
+        const req: AxiosRequestConfig = {
+            method: 'POST',
+            url,
+            headers: {
+                'x-api-key': adminApiKey,
+            }
+        };
+        if (token) {
+            req.headers.Authorization = 'Bearer ' + token;
+        }
+        return req;
+    }
+
+    abstract handlePost(agentUrl: string, agentId: string, adminApiKey: string, route: string, topic: string,
```

Comment:
Not necessary, but I think you could make this `abstract handlePost: HandlerCallback` in order to more strictly tie the definition here to the type alias.

Reply:
This is not a callback method identified with the HandlerCallback type.  This method is the signature of the API acapy expects.

Reply:
I see. It's slightly confusing that they're semantically different but functionally the same. Could probably clean that up with some different typings, but maybe not worth it right now.